api/src/main/kotlin/com/google/devtools/ksp/utils.kt

```diff
@@ -451,7 +451,15 @@ private fun List<*>.toArray(method: Method, valueProvider: (Any) -> Any): Array<
 
 @Suppress("UNCHECKED_CAST")
 private fun <T> Any.asEnum(returnType: Class<T>): T =
-    returnType.getDeclaredMethod("valueOf", String::class.java).invoke(null, this.toString()) as T
+    returnType.getDeclaredMethod("valueOf", String::class.java)
+        .invoke(
+            null,
+            if (this is KSType) {
```

Comment:
(Related discussion; Not for this PR)

Without the type checking it would result in the following error, where the type of `this` is `TestEnum`, rather than a `KSType`.
https://github.com/google/ksp/runs/3911397835#step:9:729

@neetopia Does the need of type checking here indicate a discrepancy in implementations of annotation value? In other words, shouldn't it always be `KSType`?

Reply:
For example, [this](https://github.com/google/ksp/commit/25fb02ab28a074641c5b6ef873f59181173dfd7a) was one of the difference among the implementations of .class and sources.

It could also be the reason why it's only reproducible with KSP 1.5.31-1.0.0; The `toString()`s of class instance and `KSType` are different.